plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -3149,11 +3149,7 @@ public int compare(final DiskTO arg0, final DiskTO arg1) {
                     if (volume.getType() == Volume.Type.DATADISK && !(isWindowsTemplate && isUefiEnabled)) {
                         disk.defFileBasedDisk(physicalDisk.getPath(), devId, diskBusTypeData, DiskDef.DiskFmtType.QCOW2);
                     } else {
-                        if (isSecureBoot) {
-                            disk.defFileBasedDisk(physicalDisk.getPath(), devId, DiskDef.DiskFmtType.QCOW2, isWindowsTemplate);
-                        } else {
-                            disk.defFileBasedDisk(physicalDisk.getPath(), devId, diskBusType, DiskDef.DiskFmtType.QCOW2);
-                        }
+                        disk.defFileBasedDisk(physicalDisk.getPath(), devId, diskBusType, DiskDef.DiskFmtType.QCOW2);
```

Comment:
@shwstppr 
When the uefi functionality implemented there is a issue with (Windows Guest VM + Secure Boot )Root Disk controller for boot disk using virtio, hence it's been set for windows  guest  vm -> sata, Linux  guest vm -> virtio as disk controller. 
these code changes will break the uefi functionality for windows guest vm with secure boot. 

right now I don't have setup to test the functionality, it will be good If you can test the scenario and confirm that this won't break windows guest vm along with secure boot use-case . 

And this behavior always over-rides the os configuration setting, no matter what controller configured in the OS details.  Which needs to be addressed if the above use-case is not broken. Other wise we need to find a mid-way to fix both.